cluster/addons/metadata-proxy/gce/metadata-proxy-configmap.yaml

```diff
@@ -23,37 +23,63 @@ data:
 
         # By default, return 403. This protects us from new API versions.
         location / {
-            return 403;
+            return 403 "This metadata API is not allowed by the metadata proxy.";
         }
 
         # Allow for REST discovery.
         location = / {
+            if ($args ~ "recursive=true") {
```

Comment:
have you tried if `recursive=1` or `recursive=True` or `recursive=y` etc also effective on GCE metadata server. If it's more permissive than we blacklisted here, it's still a problem.